xrpl/models/transactions/transaction.py

```diff
@@ -336,7 +337,29 @@ def to_dict(self: Transaction) -> Dict[str, Any]:
             "flags": self._flags_to_int(),
         }
 
+    def _bool_flag_to_list(self: Transaction) -> List[int]:
+        tx_flags: List[str] = [
+            attr
+            for attr, set_flag in vars(self).items()
+            if (attr.startswith(TX_FLAG_PREFIXES) and set_flag)
+        ]
+        if tx_flags:
+            return [TX_FLAGS[self.transaction_type][tx_flag] for tx_flag in tx_flags]
+        return []
+
     def _flags_to_int(self: Transaction) -> int:
+        tx_flags: List[int] = self._bool_flag_to_list()
+        if tx_flags and (
+            self.flags if isinstance(self.flags, list) else self.flags > 0
```

Comment:
This logic is a bit hard to read in my opinion. Maybe it'd be better to split it into a helper function with a descriptive name?